www/src/components/docs/outdatedDocsBanner.astro

```diff
@@ -0,0 +1,65 @@
+---
+import { fetchGithub } from "../../utils/fetchGithub";
+
+export interface Props {
+  path: string;
+}
+const { path } = Astro.props;
+
+const [_1, _2, _3, ...rest] = path.split("/");
+
+const englishPath = `src/pages/en/${rest.join("/")}`;
+const engHref = `/en/${rest}`.replace(/(.*)\.[^.]+$/, "");
+
+const url = `https://api.github.com/repos/t3-oss/create-t3-app/commits?path=www/${path}&per_page=1`;
+const engUrl = `https://api.github.com/repos/t3-oss/create-t3-app/commits?path=www/${englishPath}&per_page=1`;
+
+// File may not be on Github yet, so we only throw an error if we're in production
+const isProduction = process.env.NODE_ENV === "production";
+const thisLangCommit = await fetchGithub(url, {
+  throwIfNoAuth: isProduction,
+  throwIfNotOk: isProduction,
+});
+const englishCommit = await fetchGithub(engUrl, {
+  throwIfNoAuth: isProduction,
+  throwIfNotOk: isProduction,
+});
+
+const thisLangDate = thisLangCommit.length
+  ? new Date(thisLangCommit[0].commit.author.date)
+  : new Date();
+const englishDate = englishCommit.length
+  ? new Date(englishCommit[0].commit.author.date)
+  : new Date();
+
+const diffInDays = Math.round(
+  Math.abs(thisLangDate.getTime() - englishDate.getTime()) / (1000 * 3600 * 24),
+);
+
+const ONE_WEEK = 1000 * 60 * 60 * 24 * 7;
+Astro.response.headers.set("Cache-Control", `s-max-age=${ONE_WEEK}`);
+---
+
+{
+  diffInDays > 0 && (
+    <div class="mb-8 w-full rounded-lg bg-t3-purple-200/50 p-4 font-medium dark:bg-t3-purple-200/10">
+      <p>
+        Attention: This page is{" "}
+        <span class="font-extrabold">
+          {diffInDays} {diffInDays === 1 ? "day" : "days"}
+        </span>
+        older than the English version, and might be outdated. If you're a
```

Comment:
could this take a `language: KnownLanguage` prop (or some other way to know what the active language is) and then render the message in that language, with a fallback to English?

Reply:
oh probably just `const [_1, _2, lang] = path.split("/");`